probes/pinsDependencies/impl.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2024 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//nolint:stylecheck
+package pinsDependencies
+
+import (
+	"embed"
+	"fmt"
+
+	"github.com/ossf/scorecard/v4/checker"
+	"github.com/ossf/scorecard/v4/checks/fileparser"
+	sce "github.com/ossf/scorecard/v4/errors"
+	"github.com/ossf/scorecard/v4/finding"
+	"github.com/ossf/scorecard/v4/finding/probe"
+	"github.com/ossf/scorecard/v4/probes/internal/utils/uerror"
+	"github.com/ossf/scorecard/v4/rule"
+)
+
+//go:embed *.yml
+var fs embed.FS
+
+const (
+	Probe      = "pinsDependencies"
+	DepTypeKey = "dependencyType"
+)
+
+func Run(raw *checker.RawResults) ([]finding.Finding, string, error) {
+	if raw == nil {
+		return nil, "", fmt.Errorf("%w: raw", uerror.ErrNil)
+	}
+
+	var findings []finding.Finding
+
+	r := raw.PinningDependenciesResults
+
+	for i := range r.ProcessingErrors {
+		e := r.ProcessingErrors[i]
+		f := finding.Finding{
+			Message:  generateTextIncompleteResults(e),
+			Location: &e.Location,
+			Outcome:  finding.OutcomeError,
+		}
```

Comment:
here, and below, should use the various `finding.New` functions. This one doesn't set the `Probe`, so you get runtime errors like:
```
check runtime error: Pinned-Dependencies: internal error: invalid probe results
```

Reply:
Should all be fixed.